ios/FluentUI/List/ListActionItem.swift

```diff
@@ -0,0 +1,199 @@
+        }
+
+        @ViewBuilder
+        var secondaryButton: some View {
+            if let secondaryAction {
+                SwiftUI.Button(secondaryAction.title) {
+                    secondaryAction.handler()
+                }
+                .buttonStyle(ActionButtonStyle(actionType: secondaryAction.actionType, tokenSet: tokenSet))
+                .frame(maxWidth: .infinity, alignment: .center)
+            }
+        }
+
+        @ViewBuilder
+        var separatorsStack: some View {
+            VStack {
+                if let topSeparatorType {
+                    SeparatorRepresentable(orientation: .horizontal)
+                        .frame(height: Separator.thickness)
+                        .padding(edgeInsets(for: topSeparatorType))
+                }
+                Spacer()
+                if let bottomSeparatorType {
+                    SeparatorRepresentable(orientation: .horizontal)
+                        .frame(height: Separator.thickness)
+                        .padding(edgeInsets(for: bottomSeparatorType))
+                }
+            }
+            .accessibilityHidden(true)
+        }
+
+        @ViewBuilder
+        var buttons: some View {
+            if secondaryAction != nil {
+                HStack {
+                    primaryButton
+                        .padding(.leading, ListItemTokenSet.paddingLeading)
+                        .padding(.trailing, ListItemTokenSet.horizontalSpacing)
+                    SeparatorRepresentable(orientation: .horizontal)
+                        .frame(width: Separator.thickness)
+                    secondaryButton
+                        .padding(.leading, ListItemTokenSet.horizontalSpacing)
+                        .padding(.trailing, ListItemTokenSet.paddingTrailing)
+                }
+            } else {
+                primaryButton
+                    .padding(.leading, ListItemTokenSet.paddingLeading)
+                    .padding(.trailing, ListItemTokenSet.paddingTrailing)
+            }
+        }
+
+        @ViewBuilder
+        var backgroundView: some View {
+            if let backgroundColor = backgroundStyleType.defaultColor(tokenSet: tokenSet) {
+                Color(backgroundColor)
+            } else {
+                EmptyView()
+            }
+        }
+
+        @ViewBuilder
+        var content: some View {
+            if topSeparatorType != nil || bottomSeparatorType != nil {
+                ZStack {
+                    buttons
+                    separatorsStack
+                }
+            } else {
+                buttons
+            }
+        }
+
+        return content
+            .listRowInsets(EdgeInsets())
+            .listRowSeparator(.hidden)
+            .frame(minHeight: ListItemTokenSet.oneLineMinHeight)
+            .background(backgroundView)
+    }
+
+    /// The  type of separator on the top edge
+    var topSeparatorType: ListActionItemSeparatorType?
```

Comment:
Seems odd to have top and bottom separator types as optional. Shouldn't there be a default type instead just like `backgroundStyleType`?

Reply:
hmm yeah, I am not sure. It can kinda go either way depending on if you are using grouped sections in list (which has separators) or a plain style which doesnt.

not that it is the correct way, but I followed the pattern of `ActionsCell`, which had then set to none with the table style as plain:
``` swift    
/// Style describing whether or not the cell's custom top separator should be visible and how wide it should extend
    @objc open var topSeparatorType: TableViewCell.SeparatorType = .none {
        didSet {
            if topSeparatorType != oldValue {
                updateHorizontalSeparator(topSeparator, with: topSeparatorType)
            }
        }
    }
    /// Style describing whether or not the cell's custom bottom separator should be visible and how wide it should extend
    @objc open var bottomSeparatorType: TableViewCell.SeparatorType = .none {
        didSet {
            if bottomSeparatorType != oldValue {
                updateHorizontalSeparator(bottomSeparator, with: bottomSeparatorType)
            }
        }
    }
```

Reply:
I think to make it more consistent with the pattern of ActionsCell we'd need to add the default none style and make it non-optional. It's not a big deal though so it's your call.

Reply:
I guess I dont agree with having an enum for `.none`.  Is that not the goal of having optional values in swift? (`Optional.none` which is the same as nil) As for this scenario we dont show something in the view at all, its not a modification of a view that will be shown.

That was the reasoning on why I made a new `enum` for this which only has `.inset` and `.full`, instead of following the pattern of having a "none" case for each enum.

Thoughts?

Reply:
I would agree if this was an optional field for the separator itself. However, an _optional type_ reads as if sometimes we have a separator type and sometimes we don't which semantically sounds wrong to me. Also this is the practice we usually adopt in fluent where we routinely use none in enum cases to show a missing element (i.e. DrawerResizingBehavior, AvatarActivityStyle, AvatarPresence, TwoLineTitleViewAccessoryType etc.). It also seems to be common practice within UIKit as well.

Reply:
updated to reflect this.